specification/azsadmin/resource-manager/storage/Microsoft.Storage.Admin/preview/2019-08-08/storageaccounts.json

```diff
@@ -0,0 +1,593 @@
+          "readOnly": true
+        },
+        "primaryLocation": {
+          "description": "The primary location for the storage account.",
+          "type": "string",
+          "readOnly": true
+        },
+        "faultDomain": {
+          "description": "The fault domain for the storage account.",
+          "type": "string",
+          "readOnly": true
+        },
+        "statusOfPrimary": {
+          "description": "The status of primary location of storage account.",
+          "$ref": "storageaccounts.json#/definitions/LocationStatus",
+          "readOnly": true
+        },
+        "tenantSubscriptionId": {
+          "description": "Subscription ID of the subscription under which the storage account locates.",
+          "type": "string",
+          "readOnly": true
+        },
+        "tenantStorageAccountName": {
+          "description": "Storage account name from tenant view.",
+          "type": "string",
+          "readOnly": true
+        },
+        "tenantResourceGroupName": {
+          "description": "The name of resource group under which the storage account locates.",
+          "type": "string",
+          "readOnly": true
+        },
+        "deletedTime": {
+          "description": "The date-time when the storage account was deleted.",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "accountStatus": {
+          "$ref": "storageaccounts.json#/definitions/StorageAccountStatus",
+          "description": "The status of storage account."
+        },
+        "permissions": {
+          "description": "The permissions on the storage account.",
+          "$ref": "storageaccounts.json#/definitions/WacAccountPermissions",
+          "readOnly": true
+        },
+        "accountId": {
+          "description": "Internal storage account ID, which is not visible to tenant.",
+          "type": "string",
+          "readOnly": true
+        },
+        "wacInternalState": {
+          "description": "The internal state of storage account in WAC.",
+          "$ref": "storageaccounts.json#/definitions/WacAccountStates",
+          "readOnly": true
+        },
+        "encryption": {
+          "description": "The storage encryption setting",
+          "$ref": "storageaccounts.json#/definitions/StorageEncryption",
+          "readOnly": true
+        },
+        "supportsHttpsTrafficOnly": {
+          "description": "Storage account supports https traffic only or not",
+          "type": "boolean",
+          "readOnly": true
+        }
+      }
+    },
+    "StorageAccount": {
+      "description": "Properties of the storage account.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "Properties of the storage account.",
+          "x-ms-client-flatten": true,
+          "$ref": "storageaccounts.json#/definitions/StorageAccountProperties",
+          "readOnly": true
+        },
+        "name": {
```

Comment:
Why did you expand the type /resource inline? It was:

"allOf": [
        {
          "$ref": "storage.json#/definitions/Resource"
        }
      ]

We recommend using a type reference to bring in id, name, type... This makes it easier to verify all resources have the correct schema.

Reply:
Will extract one for storage account and quota.